### ISSUE TITLE
Recipe - Align Jackson dependencies using the BOM

### ISSUE THREAD

**Issue by yeikel** [openrewrite/rewrite#2884]
https://github.com/FasterXML/jackson-bom

Can we create a generic recipe to replace manual dependencies with a BOM?

Related issue https://github.com/openrewrite/rewrite-testing-frameworks/issues/318

**Comment by sambsnyd** [comment]
Yes, creating such a recipe is definitely possible. The coordinates of the bom could be specified to the recipe, which would download and resolve the bom. Then knowing which dependencies it manages, it would know what version numbers to remove. Would be very handy.

**Comment by yeikel** [comment]
> Yes, creating such a recipe is definitely possible. The coordinates of the bom could be specified to the recipe, which would download and resolve the bom. Then knowing which dependencies it manages, it would know what version numbers to remove. Would be very handy.



> Yes, creating such a recipe is definitely possible. The coordinates of the bom could be specified to the recipe, which would download and resolve the bom. Then knowing which dependencies it manages, it would know what version numbers to remove. Would be very handy.

Thank you for confirming. I created https://github.com/openrewrite/rewrite/issues/2887

This and #318 could be converted to YAML-only recipes afterwards for convenience

**Comment by yeikel** [comment]
> Yes, creating such a recipe is definitely possible. The coordinates of the bom could be specified to the recipe, which would download and resolve the bom. Then knowing which dependencies it manages, it would know what version numbers to remove. Would be very handy.

@sambsnyd Should I create other BOM related recipes as separate issues? To be implemented in the future using YAML-only recipes

For example there are a couple that should be useful:

- Koltin bom
- LogJ bom
